### PR TITLE
Moving the System.Globalization contract into System.Runtime

### PR DESCRIPTION
**Pull request by sokket**
Part one of the globalization work; moving the contract into System.Runtime

/cc: @weshaggard @joperezr 
/fyi: @danmosemsft

### CODE REVIEW

File: src/System.Globalization/ref/System.Globalization.cs [hunk 1]
```diff
@@ -15,88 +15,6 @@ public static partial class CharUnicodeInfo
         public static System.Globalization.UnicodeCategory GetUnicodeCategory(char ch) { return default(System.Globalization.UnicodeCategory); }
         public static System.Globalization.UnicodeCategory GetUnicodeCategory(string s, int index) { return default(System.Globalization.UnicodeCategory); }
     }
-    public partial class CompareInfo
```
Comment (joperezr) [inline src/System.Globalization/ref/System.Globalization.cs]: >public partial class CompareInfo [](start = 4, length = 32)

You will have to manually add type forwards to all of these types you are pushing down to System.Runtime. You can see an example for this in #10343

**Comment by joperezr** [comment]
Another way of doing this "cleaner" would be to first add the types on System.Runtime, and then on a later commit remove them from globalization. Something like I did here in #10241

File: src/System.Globalization/ref/System.Globalization.cs (hunk 1 above)
Comment (sokket) [inline src/System.Globalization/ref/System.Globalization.cs]: Added

File: src/System.Runtime/ref/System.Runtime.cs [hunk 2]
```diff
@@ -3356,3 +3356,231 @@ public abstract partial class WaitHandle : System.IDisposable
         public virtual bool WaitOne(System.TimeSpan timeout) { return default(bool); }
     }
 }
+
+namespace System.Globalization
```
Comment (joperezr) [inline src/System.Runtime/ref/System.Runtime.cs]: There already is a `namespace System.Globalization` declaration in this file, you should add the missing types in that one.

File: src/System.Runtime/ref/System.Runtime.cs [hunk 3]
```diff
@@ -3356,3 +3356,231 @@ public abstract partial class WaitHandle : System.IDisposable
...
+{
+    public enum CalendarWeekRule
```
Comment (joperezr) [inline src/System.Runtime/ref/System.Runtime.cs]: This guy is there already

File: src/System.Runtime/ref/System.Runtime.cs [hunk 4]
```diff
@@ -3356,3 +3356,231 @@ public abstract partial class WaitHandle : System.IDisposable
...
+    {
+        FirstDay = 0,
+        FirstFourDayWeek = 2,
+        FirstFullWeek = 1,
+    }
+    public abstract partial class Calendar
```
Comment (joperezr) [inline src/System.Runtime/ref/System.Runtime.cs]: Same here, this one should already be in this file, in the other namespace declaration for System.Globalization

File: src/System.Globalization/ref/System.Globalization.cs [hunk 5]
```diff
@@ -15,88 +15,6 @@ public static partial class CharUnicodeInfo
...
-        public virtual System.Globalization.TextInfo TextInfo { get { return default(System.Globalization.TextInfo); } }
-        public virtual string TwoLetterISOLanguageName { get { return default(string); } }
-        public virtual object Clone() { return default(object); }
-        public override bool Equals(object value) { return default(bool); }
-        public virtual object GetFormat(System.Type formatType) { return default(object); }
-        public override int GetHashCode() { return default(int); }
-        public static System.Globalization.CultureInfo ReadOnly(System.Globalization.CultureInfo ci) { return default(System.Globalization.CultureInfo); }
-        public override string ToString() { return default(string); }
-    }
     public partial class CultureNotFoundException : System.ArgumentException
```
Comment (weshaggard) [inline src/System.Globalization/ref/System.Globalization.cs]: This should be moved down with CultureInfo.

File: src/System.Globalization/ref/System.Globalization.cs [hunk 6]
```diff
@@ -15,88 +15,6 @@ public static partial class CharUnicodeInfo
         public static System.Globalization.UnicodeCategory GetUnicodeCategory(char ch) { return default(System.Globalization.UnicodeCategory); }
```
Comment (weshaggard) [inline src/System.Globalization/ref/System.Globalization.cs]: CharUnicodeInfo and UnicodeCategory can be moved down as they will be exposed by Char.GetUnicodeCategory methods

File: src/System.Globalization/ref/System.Globalization.Forwards.cs [hunk 7]
```diff
@@ -5,3 +5,9 @@
 // These two types were moved down to System.Runtime
 [assembly: System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Globalization.Calendar))]
 [assembly: System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Globalization.CalendarWeekRule))]
+[assembly: System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Globalization.CompareOptions))]
```
Comment (joperezr) [inline src/System.Globalization/ref/System.Globalization.Forwards.cs]: In order to be able to resolve all of these new forwards, you will need to add a reference to System.Runtime.pkgproj to the ref/System.Globalization.csproj. Alternatively you can keep these guys in System.Globalization temporarily until we publish a System.Runtime that has your change, and then remove them here.

File: src/System.Globalization/ref/System.Globalization.cs (hunk 6 above)
Comment (weshaggard) [inline src/System.Globalization/ref/System.Globalization.cs]: Sorry just UnicodeCategory will be exposed by Char. CharUnicodeInfo is separate, at least for now.

**Comment by joperezr** [comment]
@weshaggard I probably should have asked you this when I moved Calendar and CalendarWeekRule down, but what is the plan for tests of types that are moving to a different contract? Do we keep them in the old to test them in netcoreapp1.0 or do we move them to the new project and test them against 1.1?

**Comment by weshaggard** [comment]
>@weshaggard I probably should have asked you this when I moved Calendar and CalendarWeekRule down, but what is the plan for tests of types that are moving to a different contract? Do we keep them in the old to test them in netcoreapp1.0 or do we move them to the new project and test them against 1.1?

That's a great question. For now lets try to keep them where they are but we will likely end up moving them in the future.

**Comment by ellismg** [comment]
Jon, your git config is incorrect for user.name and user.email, so your commits don't appear as "owned" by you in github.  Can you fix this and then fix up the commits?

File: src/System.Runtime/ref/System.Runtime.cs (hunk 2 above)
Comment (sokket) [inline src/System.Runtime/ref/System.Runtime.cs]: Fixed

File: src/System.Runtime/ref/System.Runtime.cs (hunk 3 above)
Comment (sokket) [inline src/System.Runtime/ref/System.Runtime.cs]: Fixed

File: src/System.Runtime/ref/System.Runtime.cs (hunk 4 above)
Comment (sokket) [inline src/System.Runtime/ref/System.Runtime.cs]: Fixed

File: src/System.Globalization/ref/System.Globalization.cs (hunk 5 above)
Comment (sokket) [inline src/System.Globalization/ref/System.Globalization.cs]: Fixed

File: src/System.Globalization/ref/System.Globalization.cs (hunk 6 above)
Comment (sokket) [inline src/System.Globalization/ref/System.Globalization.cs]: Fixed

File: src/System.Globalization/ref/System.Globalization.Forwards.cs (hunk 7 above)
Comment (sokket) [inline src/System.Globalization/ref/System.Globalization.Forwards.cs]: Added

**Comment by sokket** [comment]
@ellismg Oops, sorry...nuked my machine before and forgot to reset these values. Fixed

File: src/System.Collections/tests/project.json [hunk 8]
```diff
@@ -2,7 +2,6 @@
   "dependencies": {
     "Microsoft.NETCore.Platforms": "1.0.2-beta-devapi-24401-01",
     "System.Collections": "4.0.12-beta-devapi-24401-01",
-    "System.Globalization": "4.0.12-beta-devapi-24401-01",
```
Comment (joperezr) [inline src/System.Collections/tests/project.json]: I'm not sure why you had to change this

**Comment by weshaggard** [comment]
@sokket any update on this? Please make sure you merge with @joperezr latest PR.

**Comment by sokket** [comment]
Sorry, I was working on this in the background with test cop stuff. Checking to see if my fix helps the builds pass locally, I'll be pushing an update soon

**Comment by sokket** [comment]
@joperezr @weshaggard I have green runs again :) any further comments?

File: src/System.Globalization/ref/System.Globalization.csproj [hunk 9]
```diff
@@ -14,5 +14,9 @@
   <ItemGroup>
     <None Include="project.json" />
   </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\..\System.Runtime\ref\System.Runtime.csproj">
```
Comment (weshaggard) [inline src/System.Globalization/ref/System.Globalization.csproj]: Lets have a work item tracking the removal of these ProjectReferences once we get packages with the changes.

**Comment by weshaggard** [comment]
LGTM

**Pull request merged**